### PR TITLE
Add matcher for wrapped error comparison

### PR DESCRIPTION
**Pull request by flaviamissi**
Now we can check if a wrapped error contains a given error type wrapped at some point in the stack, e.g.:

```go
originalErr := MyCustomError{message: "My original eror"}
err := errors.Wrap(originalErr, "wrapping", "for the first time")
err = errors.Wrap(err, "wrapping again", "for the second time")
errors.Matches(err, MyCustomError{}) // returns true in this example
```

### CODE REVIEW

**Comment by ottob** [comment]
lgtm, squash and merge?

File: errors.go [hunk 1]
```diff
@@ -130,6 +140,39 @@ func Values(err error) keyvals {
...
+		}
+		err = cause.cause()
+		if errorTypeMatches(err, sample) {
+			return true
+		}
+	}
+	return false
+}
+
+func errorTypeMatches(err error, tomatch error) bool {
```
Comment (hwsoderlund) [inline errors.go]: `toMatch` ?

Comment (flaviamissi) [inline errors.go]: will do

File: errors.go [hunk 2]
```diff
@@ -130,6 +140,39 @@ func Values(err error) keyvals {
 	return v
 }
 
+// Matches a given error `err` with a given type `sample` looking recursivelly through the error stack.
```
Comment (hwsoderlund) [inline errors.go]: typo: recursivelly

**Comment by flaviamissi** [comment]
Rewrote history (wow! :trollface:) and merging now!

**Pull request merged**